libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutGenerator.java

```diff
@@ -0,0 +1,113 @@
+  private final TablePartitionStats tablePartitionStats;
+
+  /**
+   * Generate a list of data layout optimization strategies based on the table file stats and
+   * historic query patterns.
+   */
+  @Override
+  public List<DataLayoutOptimizationStrategy> generate() {
+    // skip single partition and non-partitioned tables
+    if (tablePartitionStats.get().count() <= 1) {
+      return Collections.emptyList();
+    }
+    return Collections.singletonList(generateCompactionStrategy());
+  }
+
+  private DataLayoutOptimizationStrategy generateCompactionStrategy() {
+    List<Long> fileSizes =
+        tableFileStats
+            .get()
+            .map((MapFunction<FileStat, Long>) FileStat::getSize, Encoders.LONG())
+            .filter(
+                (FilterFunction<Long>)
+                    size ->
+                        size < TARGET_BYTES_SIZE * DataCompactionConfig.MIN_BYTE_SIZE_RATIO_DEFAULT)
+            .collectAsList();
+
+    DataCompactionConfig.DataCompactionConfigBuilder configBuilder = DataCompactionConfig.builder();
+
+    configBuilder.targetByteSize(TARGET_BYTES_SIZE);
+
+    long totalSizeBytes = fileSizes.stream().reduce(0L, Long::sum);
+    long estimatedFileGroupsCount =
+        Math.max(
+            tablePartitionStats.get().count(),
+            totalSizeBytes / DataCompactionConfig.MAX_FILE_GROUP_SIZE_BYTES_DEFAULT);
+
+    int maxCommitsCount =
+        (int)
+            Math.min(
+                MAX_NUM_COMMITS,
+                (estimatedFileGroupsCount + NUM_FILE_GROUPS_PER_COMMIT - 1)
+                    / NUM_FILE_GROUPS_PER_COMMIT);
+    configBuilder.partialProgressMaxCommits(maxCommitsCount);
+
+    int estimatedTasksPerFileGroupCount =
+        (int) (DataCompactionConfig.MAX_FILE_GROUP_SIZE_BYTES_DEFAULT / TARGET_BYTES_SIZE);
+
+    configBuilder.maxConcurrentFileGroupRewrites(
+        Math.min(
+            MAX_CONCURRENT_FILE_GROUP_REWRITES,
+            (estimatedTasksPerFileGroupCount + REWRITE_PARALLELISM - 1)
+                / estimatedTasksPerFileGroupCount));
+
+    // don't split large files
+    configBuilder.maxByteSizeRatio(MAX_BYTES_SIZE_RATIO);
+
+    long filesReducedCount = estimateReducedFilesCount(fileSizes);
+    double computeGbHr = estimateComputeGbHr(fileSizes);
+    double filesReducedCountPerComputeGbHr = filesReducedCount / computeGbHr;
+    return DataLayoutOptimizationStrategy.builder()
+        .config(configBuilder.build())
+        .cost(computeGbHr)
+        .gain(filesReducedCount)
+        .score(filesReducedCountPerComputeGbHr)
+        .build();
+  }
+
+  private long estimateReducedFilesCount(List<Long> fileSizes) {
+    long candidateFilesBytes = fileSizes.stream().reduce(0L, Long::sum);
+    long compactedFilesCount = (candidateFilesBytes + TARGET_BYTES_SIZE - 1) / TARGET_BYTES_SIZE;
+    return Math.max(0, fileSizes.size() - compactedFilesCount);
+  }
+
+  private double estimateComputeGbHr(List<Long> fileSizes) {
```

Comment:
@shardulm94 can you please take a look at the math in this function where we are trying to compute the estimated GBhr for the compaction job that we will run?

```
to_hour(totalBytesToWriteAsPartOfFileRightSizing / rewrite_throughput_in_bytes_per_second)
```

Reply:
Also, please confirm `COMPUTE_STARTUP_COST_GB_HR` from your observations. p50 should be a good start.

Reply:
we can start with 0.5 and tune later